rust/rbac-registration/src/cardano/cip509/mod.rs

```diff
@@ -101,43 +122,57 @@ impl Decode<'_, ()> for Cip509 {
                 let _: u8 = decode_helper(d, "CIP509", ctx)?;
                 match key {
                     Cip509IntIdentifier::Purpose => {
-                        cip509_metadatum.purpose =
-                            Uuid::try_from(decode_bytes(d, "CIP509 purpose")?).map_err(|_| {
-                                decode::Error::message("Invalid data size of Purpose")
-                            })?;
+                        purpose = Uuid::try_from(decode_bytes(d, "CIP509 purpose")?)
+                            .map_err(|_| decode::Error::message("Invalid data size of Purpose"))?;
                     },
                     Cip509IntIdentifier::TxInputsHash => {
-                        cip509_metadatum.txn_inputs_hash =
+                        txn_inputs_hash =
                             TxInputHash::try_from(decode_bytes(d, "CIP509 txn inputs hash")?)
                                 .map_err(|_| {
                                     decode::Error::message("Invalid data size of TxInputsHash")
                                 })?;
                     },
                     Cip509IntIdentifier::PreviousTxId => {
-                        let prv_tx_hash: [u8; 32] = decode_bytes(d, "CIP509 previous tx ID")?
+                        let hash: [u8; 32] = decode_bytes(d, "CIP509 previous tx ID")?
                             .try_into()
                             .map_err(|_| {
-                                decode::Error::message("Invalid data size of PreviousTxId")
-                            })?;
-                        cip509_metadatum.prv_tx_id = Some(Hash::from(prv_tx_hash));
+                            decode::Error::message("Invalid data size of PreviousTxId")
+                        })?;
+                        prv_tx_id = Some(Hash::from(hash));
                     },
                     Cip509IntIdentifier::ValidationSignature => {
-                        let validation_signature = decode_bytes(d, "CIP509 validation signature")?;
-                        if validation_signature.is_empty() || validation_signature.len() > 64 {
-                            return Err(decode::Error::message(
-                                "Invalid data size of ValidationSignature",
-                            ));
-                        }
-                        cip509_metadatum.validation_signature = validation_signature;
+                        let signature = decode_bytes(d, "CIP509 validation signature")?;
+                        validation_signature = signature;
                     },
                 }
             } else {
                 // Handle the x509 chunks 10 11 12
                 let x509_chunks = X509Chunks::decode(d, ctx)?;
-                cip509_metadatum.x509_chunks = x509_chunks;
+                // Technically it is possible to store multiple copies (or different instances) of
```

Comment:
Sorry for miscommunication, storing multiple violate the CDDL spec, so only 1 is allowed

Reply:
I think you were clear, perhaps it is a misunderstanding from my side. What I wanted to say with this comment is that it is possible to produce metadata with multiple different types, but it indeed would violate the specification.